compiler/AST/iterator.cpp

```diff
@@ -603,8 +628,13 @@ isSingleLoopIterator(FnSymbol* fn, Vec<BaseAST*>& asts) {
 
     // If the iterator  contains a goto statement, it is not considered to be a
     // single loop iterator.
-    else if (isGotoStmt(ast)) {
-      return NULL;
+    else if (GotoStmt* gt = toGotoStmt(ast)) {
+      // The exiting goto in an IBB does not affect our judgment.
```

Comment:
Is it possible for CallDestructors to add a whole block containing a Goto? I suppose it'd come up if a defer threw or returned. Probably not possible now but I'd probably still use a while loop to allow nested blocks here.

Reply:
I implemented this. I hope that it will not make `isSingleLoopIterator()` return true when it shouldn't.